test/e2e/petset.go

```diff
@@ -581,14 +721,22 @@ func (p *petSetTester) setHealthy(ps *apps.PetSet) {
 			framework.Failf("Found multiple non-healthy pets: %v and %v", pod.Name, markedHealthyPod)
 		}
 		p, err := framework.UpdatePodWithRetries(p.c, pod.Namespace, pod.Name, func(up *api.Pod) {
-			up.Annotations[petset.PetSetInitAnnotation] = "true"
+			up.Annotations[petset.PetSetInitAnnotation] = value
 		})
 		ExpectNoError(err)
 		framework.Logf("Set annotation %v to %v on pod %v", petset.PetSetInitAnnotation, p.Annotations[petset.PetSetInitAnnotation], pod.Name)
 		markedHealthyPod = pod.Name
 	}
 }
 
+func (p *petSetTester) setHealthy(ps *apps.PetSet) {
```

Comment:
I meant don't use the annotation at all to trigger "unhealthy-pet", instead: 
* bring up a petset
* trigger pet-X to fail a readiness probe (by doing the nginx idea, nginx serves file content of /tmp/foo, kubectl exec rm /tmp/foo will cause it to return 500, kubelet will probe for readiness and notice this marking the pet !ready)
* Wait till this surfaces as unready pod
* Try scaling up the petset
* Verify petset is blocked on actually unready pet 
* trigger pet-X to pass readiness probe (touch /tmp/foo)
... and so on